### PR TITLE
[DEST-503][CC-???] Doubleclick/cookie match

### PR DESCRIPTION
**Pull request by ccnixon**
Adds support for DBM cookie matching functionality.

- [DEST-503](https://segment.atlassian.net/browse/DEST-503)
- [CC-???]()

### CODE REVIEW

**Comment by ccnixon** [comment]
Lots of force pushes here. Had to debug an issue with the tests in IE and was having trouble getting SauceLabs working locally 😢

File: integrations/doubleclick-floodlight/lib/index.js [hunk 1]
```diff
@@ -28,6 +42,28 @@ var Floodlight = module.exports = integration('DoubleClick Floodlight')
...
+  };
+  // In the initialize method:
+  // Check if we should load the DoubleClick ID pixel (and only proceed if we haven't already done so).
+  if (this.options.getDoubleClickId && this.options.googleNetworkId) {
+    if (!cookie('doubleclick_id_ts')) {
+      // Load the doubleclick pixel.
+      this.load('doubleclick id', {
+        googleNetworkId: this.options.googleNetworkId,
+        segmentWriteKey: this.options.segmentWriteKey,
+        // TODO: handle userId being nulls/undefined.
```
Comment (ccnixon) [inline integrations/doubleclick-floodlight/lib/index.js]: @f2prateek - what would the concern here be if `userId` was undefined? Should this be something that's handled on Personas end?

Comment (f2prateek) [reply integrations/doubleclick-floodlight/lib/index.js]: possibly - also wasn't sure how loading the pixel would be handled with a null parameter

File: integrations/doubleclick-floodlight/lib/index.js [hunk 2]
```diff
@@ -28,6 +41,18 @@ var Floodlight = module.exports = integration('DoubleClick Floodlight')
...
 
 Floodlight.prototype.initialize = function() {
+  // In the initialize method:
+  // Check if we should load the DoubleClick ID pixel (and only proceed if we haven't already done so).
+  if (this.options.getDoubleClickId && this.options.googleNetworkId) {
+    // Load the doubleclick pixel.
+    this.load('doubleclick id', {
+      googleNetworkId: this.options.googleNetworkId,
+      segmentWriteKey: this.options.segmentWriteKey,
+      // TODO: handle userId being nulls/undefined.
```
Comment (andreiko) [inline integrations/doubleclick-floodlight/lib/index.js]: Yeah, currently we end up with `user_id=null` in img src which is confusing for idsync-service. I'll filter this value out for now.

Comment (ccnixon) [reply integrations/doubleclick-floodlight/lib/index.js]: Hey @andreiko - should we only fire the pixel if `userId` exists?

Comment (andreiko) [reply integrations/doubleclick-floodlight/lib/index.js]: Can we just remove the `user_id` parameter from the pixel url in that case?

Comment (ccnixon) [reply integrations/doubleclick-floodlight/lib/index.js]: We can but is it not required to actually facilitate the match or can you do it with `anonymousId`?

Comment (andreiko) [reply integrations/doubleclick-floodlight/lib/index.js]: yeah, it would work with just the `anonymousId`

**Comment by andreiko** [comment]
I noticed that the dev version of this integration that we installed on nightmarejs.org stopped working on July 05. My guess is that it might have been switched back to master. 

Are there any plans to merge this?

**Comment by CarlosMecha** [comment]
Taking it from here. Thanks!

**Pull request closed without merging**